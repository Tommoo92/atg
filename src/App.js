/** @jsx jsx */
import React from 'react';
import logo from '../src/atg.svg';
import {css, jsx} from '@emotion/core';
import Input from './input.js';
import Games from './games.js';
import RaceDay from './raceDay.js';
import RaceInfo from './raceInfo.js';
import 'bootstrap/dist/css/bootstrap.css';

var app;
class App extends React.Component {
  constructor(){
    super();
    app = this;

    this.state = {
      data: [],
      raceDayInfo: [],
      raceInfo: []
    }
  }
  async onGameChange(game){
    var url = "/services/racinginfo/v1/api/products/"+game;
    await fetch(url)    
    .then((response) => response.json())
    .then((responseJSON) => {
       // do stuff with responseJSON here...
       console.log(responseJSON);
       app.setState({
         data: responseJSON,
         raceDayInfo: [],
         raceInfo: []
      });
  });
}
onRaceDaySelection(id){
  var url = "/services/racinginfo/v1/api/games/"+id;
  fetch(url)    
  .then((response) => response.json())
  .then((responseJSON) => {
     // do stuff with responseJSON here...
     console.log(responseJSON);
     app.setState({
      raceDayInfo: responseJSON,
      raceInfo: []
    });
});
}
onRaceSelection(race){
  console.log(race);
  app.setState({
    raceInfo: race
  })
}
  render(){
    return (
      <div className="App" css={css`text-align:center`}>
        <nav css={css`
        background-color: rgb(0, 79, 159);
        width: 100%;
        height: 50px;
        border-bottom: 4px solid #fec005;
        padding: 15px; 
        `}>
          <img  css={css`
        height: 100%;
        float: left; 
        `} src={logo}></img>  
        </nav>
        <div className="container">
          <Input onGameChange={app.onGameChange}></Input>
          {app.state.data.length !== 0 &&
            <Games data={app.state.data} onSelection={app.onRaceDaySelection}></Games>
          }
          {app.state.raceDayInfo.length !== 0 &&
            <RaceDay raceDayInfo={app.state.raceDayInfo} onRaceSelection={app.onRaceSelection} ></RaceDay>
          }
          {app.state.raceInfo.length !== 0 &&
            <RaceInfo raceInfo={app.state.raceInfo} ></RaceInfo>
          }
        </div>
      </div>
    );
  }
}

export default App;
