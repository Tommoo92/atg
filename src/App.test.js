import React from 'react';
import App from './App.js';
import Input from './input.js';
import { create } from 'react-test-renderer'

describe('Component tests',()=>{
  test("Matches the snapshot", () => {
    const app = create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });
  test('Test data state', () => {
    let app = create(<App />)
    let instance = app.getInstance();

    expect(instance.state.raceInfo).toStrictEqual([])
  })
  test('Test data state', () => {
    let app = create(<App />)
    let instance = app.getInstance();

    expect(instance.state.data).toStrictEqual([])
  })
})