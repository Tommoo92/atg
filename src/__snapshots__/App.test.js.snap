// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component tests Matches the snapshot 1`] = `
<div
  className="App css-1azakc"
>
  <nav
    className="css-867qb3"
  >
    <img
      className="css-16ouyr5"
      src="atg.svg"
    />
  </nav>
  <div
    className="container"
  >
    <div
      className="input css-1f20m0c"
    >
      <h2>
        Välj spelform
      </h2>
      <select
        className="css-18wld8k"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Välj spel
        </option>
        <option
          value="V75"
        >
          V75
        </option>
        <option
          value="V65"
        >
          V65
        </option>
        <option
          value="V64"
        >
          V64
        </option>
        <option
          value="V4"
        >
          V4
        </option>
      </select>
    </div>
  </div>
</div>
`;
